### PR TITLE
Adds Service Accounts tab

### PR DESCRIPTION
**Pull request by samuelcostae**
**Description**
Updated UI to make use of new separate Internal/service accounts API endpoints being added to security plugin backend via PR https://github.com/opensearch-project/security/pull/2786

Differently From Internal Accounts, 'Delete' and 'Create' Buttons were removed for Service Accounts.

'Internal Users' renamed to 'Internal Accounts' to be in line with 'Service Accounts' as discussed on https://github.com/opensearch-project/security/issues/2842

**Category**
Enhancement

**Why these changes are required?**
Improve UX

What is the old behavior before changes and new behavior after changes?
All accounts had to be requested, without filtering Internal/Service

**Issues Resolved**
#2704

**Testing**
Manual test on UI

**Check List**

- [ ]  New functionality includes testing
- [ ]  New functionality has been documented
- [x]  Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by peternied** [comment]
@samuelcostae  We've got an upcoming 2.10 release and I'm reviewing outstanding pull requests, would you like this pull request to be considered for inclusion?

Some commits are missing the DCO and tests failing, is this ready to review?  Let me know and I'll start to take a look

**Comment by samuelcostae** [comment]
working on adding tests to new service account panel

File: public/apps/configuration/app-router.tsx
Comment (DarshitChanpura) [inline public/apps/configuration/app-router.tsx:55]: we can leave this as users

File: public/apps/configuration/constants.tsx
Comment (DarshitChanpura) [inline public/apps/configuration/constants.tsx:29]: Do these endpoints exist in backend?

Based on the related PR https://github.com/opensearch-project/security/pull/2786/files, it seems like it is expecting query parameters `name` & `filterBy`.

**Review by DarshitChanpura — COMMENTED** [review]
One generic comment. API_ENDPOINT variables for service accounts and internal accounts are slightly confusing to me. Can you add more details on why are they needed?

File: public/apps/configuration/constants.tsx
Comment (samuelcostae) [reply public/apps/configuration/constants.tsx:29]: those endpoints in constants.tsx are actually the api endpoints from the backend of the dashboards's plugins that are called by the react front-end, not directly related to the endpoints specified on the the OS's plugin.

So the full chain is:

OS Dashboards FrontEnd -> OS Dashboards Backend -> OS Security's plugin.

Comment (DarshitChanpura) [reply public/apps/configuration/constants.tsx:29]: aah.. well learned something new today.

**Review by DarshitChanpura — DISMISSED** [review]
Thanks @samuelcostae !

File: public/apps/configuration/panels/service-account-list.tsx
Comment (scrawfor99) [inline public/apps/configuration/panels/service-account-list.tsx:211]:
```suggestion
            error={errorFlag ? 'Load data failed, please check the console log for more details.' : ''}
```

**Review by scrawfor99 — APPROVED** [review]

**Review by DarshitChanpura — APPROVED** [review]

File: public/apps/configuration/panels/service-account-list.tsx
Comment (cwperks) [inline public/apps/configuration/panels/service-account-list.tsx:179]: Is this comment applicable to service accounts?

Comment (samuelcostae) [reply public/apps/configuration/panels/service-account-list.tsx:179]: @cwperks  "Here you list the users authenticated via an external authentication system"

I thought that this was the definition of a service account.   

Do you know where can i find  the definition?

Comment (cwperks) [reply public/apps/configuration/panels/service-account-list.tsx:179]: @samuelcostae You can find a description here (https://gist.github.com/cwperks/e756e1cead72cd511d819241a11337e8#service-account--service-account-tokens). Service account tokens are issued and authenticated by the security plugin so there are no external authentication systems at play here.

**Comment by DarshitChanpura** [comment]
Here is the final look and feel:
Screenshot 2023-09-28 at 12 12 44 PM
Screenshot 2023-09-28 at 12 16 42 PM



Thank you @samuelcostae. One final question, with this PR, we are not adding the functionality to `Create Service account`, but if one is already created we can edit update and delete it. Is that correct?

**Review by DarshitChanpura — DISMISSED** [review]
LGTM. I wouldn't mind merging once the above question is answered.

**Comment by cwperks** [comment]
@DarshitChanpura I would recommend not making changes on the frontend at the moment. Service accounts are quite nascent and used for extensions, which is not being developed on at the moment. 

Is there any plan to add an experimental label on this for the initial release?

There is text on the screen about updating roles mappings of service accounts, but we have also discussed limiting service accounts of the reserved indices of the extension/plugin that they are associated with. I'm for creating UX for this feature long term, but I think its pre-mature to add UX around service accounts.

**Comment by DarshitChanpura** [comment]
> @DarshitChanpura I would recommend not making changes on the frontend at the moment. Service accounts are quite nascent and used for extensions, which is not being developed on at the moment.
> 
> Is there any plan to add an experimental label on this for the initial release?
> 
> There is text on the screen about updating roles mappings of service accounts, but there has also been discussion around limiting service accounts to the reserved indices of the extension/plugin that they are associated with. I'm for creating UX for this feature long term, but I think its pre-mature to add UX around service accounts.

Ahh.. makes sense. What do you think about enabling the view with a feature flag too?

**Comment by samuelcostae** [comment]
@cwperks  As we discussed on the call Thursday, the feature flag won't be necessary, can you review the PR as one more is needed for the merge? Thanks

File: public/apps/configuration/panels/service-account-list.tsx
Comment (cwperks) [inline public/apps/configuration/panels/service-account-list.tsx:179]: @samuelcostae @scrawfor99 @DarshitChanpura Are service accounts limited to the reserved (system) indices that the extension is associated with or will they be able to do more? I'd like to make sure the language here reflects what these accounts would be/are capable of.

Comment (DarshitChanpura) [reply public/apps/configuration/panels/service-account-list.tsx:179]: They should be limited to the indices reserved by the associated service (e.g an extension)

File: public/apps/configuration/panels/service-account-list.tsx
Comment (DarshitChanpura) [inline public/apps/configuration/panels/service-account-list.tsx:180]: nit: `from {' '}` space after from

File: public/apps/configuration/panels/service-account-list.tsx
Comment (samuelcostae) [reply public/apps/configuration/panels/service-account-list.tsx:179]: @cwperks  Only thing I can think of that could be added is "Differently from normal accounts, Service Accounts are able to access the system_indices associated assigned to their role"

File: public/apps/configuration/panels/user-list.tsx
Comment (DarshitChanpura) [inline public/apps/configuration/panels/user-list.tsx:227]: The more I look at this change, the more I feel that we should leave this as `user` in place of `account`. But that could just be me, thoughts @samuelcostae @cwperks @scrawfor99

Comment (samuelcostae) [reply public/apps/configuration/panels/user-list.tsx:227]: I dont have a strong opinion, as I think either leaving/changing would be fine.  I think it is ok with the commited change as it is in sync with "Service Accounts", but again, not a strong opinion.

File: public/apps/configuration/panels/service-account-list.tsx
Comment (scrawfor99) [reply public/apps/configuration/panels/service-account-list.tsx:179]: This is fine as is.

File: public/apps/configuration/panels/user-list.tsx
Comment (scrawfor99) [reply public/apps/configuration/panels/user-list.tsx:227]: Either is fine.

**Comment by samuelcostae** [comment]
@DarshitChanpura  Thanks for syncing up the branch with main, it seems the Cypress tests  failures at the same occurring in the other branches. 

I think that  those errors were mentioned in the call today and already being looked at, right?

**Comment by DarshitChanpura** [comment]
Yes, Cypress failures are seem unrelated to the changes in this PR.

**Comment by peternied** [comment]
@samuelcostae Looks like there are still a number of outstanding items - can you follow up on those unresolved conversations?

**Review by DarshitChanpura — APPROVED** [review]

**Comment by samuelcostae** [comment]
Hi @cwperks Could you please review this once more? Thanks

**Comment by cwperks** [comment]
Hey @samuelcostae , it still feels a bit pre-mature to me to add UX but I will defer to @scrawfor99 and @peternied who have been more involved with the planning and delivery of this feature. For the first time a new feature is released like this, I think it would be appropriate to have an `experimental` label which implies that a future minor version can make breaking changes to the feature and that backward compatibility will not be guaranteed. 

IMO I would also like to keep the terminology `internal users` because it is prevalently used. I see why `internal accounts` would be more consistent with `service accounts`, but `internalusers` is used in API routes like this: https://opensearch.org/docs/latest/security/access-control/api/#get-user

From what I understand, in an initial release service accounts will not be very interesting from an end user perspective. They are completely restricted to a set of protected indices associated with the extension/plugin that they represent. Even if its possible to map a service account to roles, the action would not result in any new behavior granted to the service account because its restricted to a set of indices strongly. 

-----

As it pertains to this particular PR, I do see that there is an option to `Edit` the account. From what I understand, a service account is currently distinguished from a regular account because it has an `attribute` with a key of `service` and a value set to `true`. Using the UX from this PR, is it possible that an `admin` can hit the `Edit` button and `demote` the account to a regular account?

**Comment by samuelcostae** [comment]
> Using the UX from this PR, is it possible that an `admin` can hit the `Edit` button and `demote` the account to a regular account?

Correct, this is currently possible.   @peternied  Is this something we would like hide in the front-end?

**Comment by samuelcostae** [comment]
@scrawfor99  friendly ping

File: public/apps/configuration/panels/user-list.tsx
Comment (RyanL1997) [inline public/apps/configuration/panels/user-list.tsx:198]: I know in the previous thread we have been talked about this, and personally, I don't have any strong opinion for both of the namings. But I was just trying to see if there is any documentation change needed for this. CC: @DarshitChanpura @cwperks @scrawfor99 .

**Review by RyanL1997 — COMMENTED** [review]
Hi @samuelcostae, thank you for taking this on. This is looking good to me, and I just left a very minor review/question for the folks about the renaming.

File: public/apps/configuration/panels/user-list.tsx
Comment (samuelcostae) [reply public/apps/configuration/panels/user-list.tsx:198]: Good point. There are bout 25  references  to "Internal Users" in the documentation.  That would have to be updated. Might be easier to just leave as internal accounts

File: public/apps/configuration/panels/user-list.tsx
Comment (RyanL1997) [inline public/apps/configuration/panels/user-list.tsx:215]: I think this is one of the unit test failure for the formatting.
```
Insert `·of,`
```

**Review by RyanL1997 — COMMENTED** [review]
Hi @samuelcostae , this is LGTM. I will approve it if we fix the minor formatting failure of unit test. Btw the failure of integration test is due to its flakiness.

**Review by RyanL1997 — APPROVED** [review]

**Comment by samuelcostae** [comment]
@DarshitChanpura  seems you need to approve again after the changes :)

**Comment by DarshitChanpura** [comment]
@samuelcostae Could you look into Unit test failures?

**Review by DarshitChanpura — APPROVED** [review]
LGTM. @cwperks any final questions before we merge this?

**Comment by cwperks** [comment]
Same with @peternied , I don't have strong opinions so I defer to the rest of the team. 

I see that there's a column for backend roles, is that something that can be added to a service account? 

If extensions was further along, I would also recommend an admin page for extensions where security settings for extensions could be configured along with other settings.

**Comment by DarshitChanpura** [comment]
> I see that there's a column for backend roles, is that something that can be added to a service account?

If we plan to, then let's address it in a follow-up PR @samuelcostae . As I see no more blockers, I'll merge this now.

**Comment by DarshitChanpura** [comment]
@samuelcostae Could you look into 2.x backport failure

**Pull request merged**
